dev-example/src/example.test.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  createMockMachineRunner,
+  createMockState,
+  createMockStateOpaque,
+} from '@actyx/machine-runner/lib/test-utils'
+import { Bid, BidderID, Passenger, Requested } from './machines/index.js'
+import { toPrettyJSONString } from './UIMachineCommon.js'
+import { isTaxiRideCancelEnabled } from './UIMachinePassenger.js'
+
+describe('State transformation tests', () => {
+  it('should verify if state transformation as written by the consumer code is correct', () => {
+    const requestDestination = 'destination'
+    const requestPickup = 'pickup'
+    const bidTime = new Date()
+    const bidPrice = 1
+    const bidderId = 'bidderId'
+
+    const machineRunner = createMockMachineRunner(Passenger.Initial, void 0)
+    machineRunner.test.feed([
+      Requested.make({
+        destination: requestDestination,
+        pickup: requestPickup,
+      }),
+      Bid.make({
+        price: bidPrice,
+        time: bidTime.toISOString(),
+      }),
+      BidderID.make({
+        id: bidderId,
+      }),
+    ])
+
+    const auction = machineRunner.get()?.as(Passenger.Auction)
+    expect(auction).toBeTruthy()
+    if (!auction) throw new Error('no auction')
```

Comment:
This seems a bit clunky and the feedback in case of errors will not include the wrong state name. We should not make our code jest-specific, so how about adding an assertion method that casts to a desired state and throws a descriptive exception including the wrong state otherwise?

Reply:
> how about adding an assertion method that casts to a desired state and throws a descriptive exception including the wrong state otherwise?

Yes, we can. But I am going to check if we can make it so that Jest and/or any other testing tools report the correct line of code when it throws.

Reply:
We can have a custom assert. It looks good, except the reporting isn't quite optimal

```
machineRunner.test.assertAs(Passenger.Ride)

machineRunner.test.assertAs(Passenger.Auction, (auction) => {
  expect(auction.payload.bids.at(0)).toEqual({
    bidderID: bidderId,
    time: bidTime,
    price: bidPrice,
  } as BidData)
})
```

![image](https://user-images.githubusercontent.com/11693412/231713742-e482191e-0d52-46fc-b8f5-a774f55ace85.png)

Reply:
Custom assert added per https://github.com/Actyx/machines/pull/30/commits/093563d4b9c378381826866301839e5627e5aca3